namerd/storage/etcd/src/main/scala/io/buoyant/namerd/storage/etcd/EtcdDtabStore.scala

```diff
@@ -1,23 +1,33 @@
 package io.buoyant.namerd
 package storage.etcd
 
-import com.twitter.finagle.Dtab
+import com.twitter.finagle.{Dtab, Path}
 import com.twitter.io.Buf
 import com.twitter.util.{Activity, Future}
-import io.buoyant.etcd.Key
+import io.buoyant.etcd.{ApiError, Key}
 
-class EtcdDtabStore(key: Key) extends DtabStore {
-  import DtabStore.Version
+class EtcdDtabStore(root: Key) extends DtabStore {
+  import DtabStore.{DtabVersionMismatchException, Version}
 
   def list(): Activity[Set[Ns]] = ???
 
   def create(ns: Ns, dtab: Dtab): Future[Unit] = ???
 
   def delete(ns: Ns): Future[Unit] = ???
 
-  def update(ns: Ns, dtab: Dtab, version: Version): Future[Unit] = ???
-
-  def put(ns: Ns, dtab: Dtab): Future[Unit] = ???
+  def update(ns: Ns, dtab: Dtab, version: Version): Future[Unit] = {
+    val Buf.Utf8(vstr) = version
+    val index = vstr.toLong
+    val buf = Buf.Utf8(dtab.show)
+    val key = root.key(Path.Utf8(ns))
+    key.compareAndSwap(buf, prevIndex = Some(index)).unit.rescue {
+      case ApiError(ApiError.TestFailed, _, _, _) =>
```

Comment:
What if the key does not exist?  We should return a DtabNamespaceDoesNotExistException in that case, I think.